prometheus/metrics.go

```diff
@@ -350,13 +347,19 @@ func replaceInvalidCharacters(metricName string) string {
 }
 
 func getNamespaceRequestRates(api v1.API, namespace string, ratesInterval string) (model.Vector, model.Vector, error) {
-	lblIn := fmt.Sprintf(`reporter="destination",destination_workload_namespace="%s"`, namespace)
+	reporter := "source"
+	if config.Get().IstioNamespace == namespace {
+		reporter = "destination"
+	}
+
+	// traffic originating outside the namespace to destinations inside the namespace
```

Comment:
I'm not really sure about the changes here. If you look at where this function is called (health.go), `getNamespaceRequestRates` is expected to return one vector for all inbound requests and another for all outbound requests, and then it filters the results to get only what is relevant. It doesn't expect to return only traffic originating from other namespace, for inbound requests.

Anyway, you can leave it as is; that's something that anyway will conflict with my PR https://github.com/kiali/kiali/pull/394/files so I'll look again over the changes and will check with live data if the health is as expected. Currently we cannot verify it since this health check is broken til my PR is merged.

Reply:
Ah, I think I was looking at this with my graph hat on.  OK, I'm going to merge but you can change as needed in your PR.